lib/iris/util.py

```diff
@@ -1554,3 +1554,91 @@ def _meshgrid(*xi, **kwargs):
         if mxii.dtype != xii.dtype:
             mxi[i] = mxii.astype(xii.dtype)
     return mxi
+
+
+def find_discontiguities(coord, abs_tol=1e-4):
+    """
+    Searches coord for discontiguities in the bounds array and masks the data
+    at those points if specified in kwargs.
+
+    Args:
+        coord:
+        An instance of :class:`iris.coords.Coord`
+
+    Kwargs:
+        abs_tol:
+        absolute tolerance in degrees
+
+    Returns:
+        Dictionary of values representing data array indices where
+        discontiguities occur in the x-direction, y-direction or both.
+    """
+    _, diffs_x, diffs_y = iris.coords._discontiguity_in_2d_bounds(coord.bounds,
+                                                                  abs_tol)
+
+    # Find coord array indices of discontinuities
+    indices_x_direction = np.where(diffs_x > abs_tol)
+    indices_y_direction = np.where(diffs_y > abs_tol)
+
+    indices_x_direction = [el.tolist() for el in indices_x_direction]
+    indices_y_direction = [el.tolist() for el in indices_y_direction]
+
+    discontiguities = {}
+
+    if len(indices_x_direction[0]) > 0:
+        # Add 1 to x-direction as the discontiguous point is to the right of
+        # the discontiguous bound
+        point_indices_x = [indices_x_direction[0],
+                           [el + 1 for el in indices_x_direction[1]]]
+        discontiguity_array_x = list(zip(*point_indices_x))
+        discontiguities['x-direction discontiguity at coord point(s)'] = \
+            discontiguity_array_x
+
+    if len(indices_y_direction[0]) > 0:
+        # Add 1 to y-direction as the discontiguous point is below the
+        # discontiguous bound
+        point_indices_y = [[el + 1 for el in indices_y_direction[0]],
+                           indices_y_direction[1]]
+        discontiguity_array_y = list(zip(*point_indices_y))
+        discontiguities['y-direction discontiguity at coord point(s)'] = \
+            discontiguity_array_y
+
+    try:
+        return discontiguities
+    except ValueError:
+        return None
+
+
+def mask_discontiguities(coord, data, abs_tol=1e-4):
```

Comment:
A boolean array would be fine for this because we don't need anything to be human-readable; we are just doing operations.  But I still disagree with creating a boolean array for the previous.

Reply:
Also what do you mean by 'this routine wouold not be needed?' 

This routine is pretty much as simple as it gets?  The whole point of having the two separate routines is that one allows you to check which points are discontiguous (as a human) and look at them and think about them, and the other allows you to just mask them without thought.  Those are both valid options and it seems stupid to just not offer one of them because that's simpler.